### PR TITLE
Issue8

### PR DESCRIPTION
**Pull request by etnalubma**
Solution for issue #8 
Breakpoints are destroyed on instance finalization in context using finalizationCallback. Safeness is guaranteed because of the use of shared_ptr. 
Instance is destroyed also using finalizationCallback. This is safe because Proxy destructor waits for thread finalization, making it safe. Note that this callback runs asynchronously, in order to let others finalization callbacks to execute and finish proxy main loop execution.

A common incremental container was implemented to reuse context containers behaviour

### CODE REVIEW

**Comment by lboquillon** [comment]
ship it

File: src/commands/src/RaiseLocalCommand.cpp [hunk 1]
```diff
@@ -4,7 +4,8 @@
  *                  Taller Technologies.
  *
  * @file        RaiseLocalCommand.cpp
- * @author      Emanuel Bringas, Leonardo Boquillón
+ * @author      Emanuel Bringas
+ * @author      Leonardo Boquillón
```
Comment (emabringas) [inline src/commands/src/RaiseLocalCommand.cpp:8]: add yourself to the author list,

File: src/gdbProxy/gdbProxy/GdbBaseAspect.h [hunk 2]
```diff
@@ -198,7 +200,8 @@ void GdbBaseAspect<Type>::continueExecution()
 template<class Type>
 void GdbBaseAspect<Type>::waitForFinalization()
 {
-    thread.join();
+    if(thread.joinable())
```
Comment (emabringas) [inline src/gdbProxy/gdbProxy/GdbBaseAspect.h:203]: add curly braces

**Comment by etnalubma** [comment]
changes from review done

**Comment by emabringas** [comment]
ship it

**Pull request merged**